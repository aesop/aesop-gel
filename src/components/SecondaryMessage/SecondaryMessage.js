import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { arrayOfLength } from './SecondaryMessage.utils';
import Message from './components/Message';
import styles from './SecondaryMessage.module.css';

const SecondaryMessage = ({ className, id, items }) => {
  if (!items || items.length === 0) return null;

  const hasMultipleMessages = items.length > 1;
  const classSet = cx(styles.base, className);

  return (
    <aside className={classSet} id={id}>
      {items.map((item, index) => {
        const { className, id, ...props } = item;

        return (
          <Message
            className={cx(className, {
              [styles.multiple]: hasMultipleMessages,
            })}
            key={id || index}
            {...props}
          />
        );
      })}
    </aside>
  );
};

SecondaryMessage.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  items: arrayOfLength.bind(null, 2),
};

SecondaryMessage.defaultProps = {
  className: undefined,
  id: undefined,
  items: undefined,
};

export default SecondaryMessage;
