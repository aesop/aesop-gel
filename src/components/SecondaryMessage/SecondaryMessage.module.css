@import 'styles/index';

.base {
  display: flex;
  flex-wrap: wrap;
  font-family: var(--font-family-sans-serif-regular);

  &,
  & * {
    box-sizing: border-box;
  }

  @media (--viewport-sm-only) {
      background-color: #f3f1e4;
    }
}

.wrapper {
  @media (--viewport-md) {
    padding-bottom: 60px;
  }
  @media (--viewport-lg) {
    padding-bottom: 100px;
  }
}

.multiple {
  @media (--viewport-md) {
    width: 50%;
  }

  &:first-of-type {
    @media (--viewport-sm-only) {
      padding-bottom: 20px;
      border-bottom: 1px solid var(--color-grey-57);
      margin-bottom: 0;

      /* border-image: linear-gradient(0.25turn, transparent, var(--color-grey-45), transparent) 10 20%; */
    }

    @media (--viewport-md) {
      border-right: 1px solid var(--color-grey-57);

      /* border-image: linear-gradient(transparent, var(--color-grey-45), transparent) 10 100%; */
    }
  }
}
