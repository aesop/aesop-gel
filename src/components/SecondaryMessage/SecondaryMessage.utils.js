const arrayOfLength = (maxLength, props, propName, componentName) => {
  const arrayPropLength = props[propName].length;

  if (arrayPropLength > maxLength) {
    return new Error(
      `Invalid array length ${arrayPropLength} (expected less than ${maxLength}) for prop ${propName} supplied to ${componentName}.`,
    );
  }
};

export { arrayOfLength };
