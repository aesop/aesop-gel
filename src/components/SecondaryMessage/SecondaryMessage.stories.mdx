import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import SecondaryMessage from './SecondaryMessage';
import SecondaryMessageFixture from './SecondaryMessage.fixture';

<Meta title="Components/SecondaryMessage" component={SecondaryMessage} />

# SecondaryMessage

<Canvas>
  <Story name="Base component">
    <SecondaryMessage {...SecondaryMessageFixture} />
  </Story>
</Canvas>

## API

<ArgsTable of={SecondaryMessage} />
