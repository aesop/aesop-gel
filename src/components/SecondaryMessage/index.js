import SecondaryMessage from './SecondaryMessage';

export default SecondaryMessage;
