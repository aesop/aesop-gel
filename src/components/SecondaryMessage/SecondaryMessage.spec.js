import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import SecondaryMessage from './SecondaryMessage';
import SecondaryMessageFixture from './SecondaryMessage.fixture';

configure({ adapter: new Adapter() });

describe('<SecondaryMessage />', () => {
  it('should be defined', () => {
    expect(SecondaryMessage).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer
      .create(<SecondaryMessage {...SecondaryMessageFixture} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
