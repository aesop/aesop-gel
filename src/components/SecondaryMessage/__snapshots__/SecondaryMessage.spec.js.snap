// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecondaryMessage /> renders base component correctly 1`] = `
<aside
  className="base exercitationCupidatat"
  id="exercitationCupidatat"
>
  <div
    className="base exercitationCupidatat multiple"
  >
    <a
      className="base left dark link"
      href="/"
      target="_self"
      title="Irure anim labore incididunt lorem occaecat"
    >
      <div
        className="content"
      >
        <h4
          className="base xXSmall dark heading"
        >
          Irure anim labore
        </h4>
        <div
          className="copy"
        >
          Irure anim labore incididunt lorem occaecat Cillum reprehenderit Duis commodo ipsum sunt fugiat qui nulla dolor nostrud tempor quis non.
        </div>
      </div>
    </a>
  </div>
  <div
    className="base exercitationCupidatat multiple"
  >
    <div
      className="content"
    >
      <h4
        className="base xXSmall dark heading"
      >
        Proident et culpa
      </h4>
      <div
        className="copy"
      >
        Proident et culpa mollit fugiat irure anim labore incididunt Lorem occaecat. Ut culpa laborum mollit sunt elit et id tempor.
      </div>
    </div>
  </div>
</aside>
`;
